### DISCUSSION TITLE
"rg a | wc -l" on stable data, changing output

### DISCUSSION THREAD

**Discussion by shoffmeister** [BurntSushi/ripgrep#2506] [Q&A]
I am confused by the results of these two runs of ripgrep:
```
~/git-research 
❯ rg a | wc -l 
27231956

~/git-research took 3s 
❯ rg a | wc -l 
27220249
```

I'd expect the total line count matching - regardless of the order of input files (cf #152). I would assume that the fragment that ripgrep emits _per file_ should always be the same. Given that nothing in `git-research` changed, omitting `--sort-files` would only result in the exact same output fragments be emitted in an undefined (random) order?

FWIW, `--sort-files` seems to always produce matching line numbers (at the cost of 6 seconds of runtime, single-threaded vs 3 seconds for parallel execution).

What am I missing?

**Comment by BurntSushi** [comment]
Dunno. It's impossible to say without providing a way for me to test it. If you can't do that, then it seems like you could do an investigation to figure out the exact differences. Some combination of the following is just what I would do for example:

1. See if you can reproduce on sub-directories in order to make the corpus smaller.
2. Save the output of ripgrep somewhere. Sort it. Then diff it to see what changed.

Your directory looks big. Maybe something is changing that you don't expect?

**Reply by shoffmeister** [reply]
That innocently looking directory is a container for a rather large number of (at times large) git repos.

I'll see what I can figure out - many thanks for the input!
